pkg/asset/agent/manifests/agentpullsecret_test.go

```diff
@@ -156,7 +156,7 @@ metadata:
   namespace: cluster-0
 stringData:
   .dockerconfigjson: 'foo'`,
-			expectedError: "invalid PullSecret configuration: StringData: Invalid value: \"foo\": invalid character 'o' in literal false (expecting 'a')",
+			expectedError: "invalid PullSecret configuration: Data: Invalid value: \"foo\": invalid character 'o' in literal false (expecting 'a')",
```

Comment:
This is definitely wrong because the invalid data is in the stringData field, not Data. As noted above, I think we should change the field to use in errors when data is missing altogether to `stringData`.

We also need a separate test for when the invalid data is actually in the `data` field.